hs_core/models.py

```diff
@@ -1349,14 +1363,29 @@ class Meta:
         unique_together = ("content_type", "object_id")
 
 def get_path(instance, filename):
-    return os.path.join(instance.content_object.short_id, 'data', 'contents', filename)
+    if instance.content_object.resource_federation_path:
+        return os.path.join(instance.content_object.resource_federation_path, instance.content_object.short_id, 'data',
+                            'contents', filename)
+    else:
+        return os.path.join(instance.content_object.short_id, 'data', 'contents', filename)
 
 class ResourceFile(models.Model):
```

Comment:
@pkdash I may not get what you are saying here, but I don't see doing so could simplify or abstract away the need to know which iRODS zone the file exists from the rest of the code base.